### ISSUE TITLE
bug in generating random variables from multivariate student's t distribution

### ISSUE THREAD

**Issue by afniedermayer** [JuliaStats/Distributions.jl#333]
```julia
julia> td=MvTDist(3.,[.0,.0],PDMat([1. 0.;0. 1.]))
GenericMvTDist{PDMat}(
df: 3.0
dim: 2
zeromean: false
μ: [0.0,0.0]
Σ: PDMat(2,2x2 Array{Float64,2}:
 1.0  0.0
 0.0  1.0,Cholesky{Float64} with factor:
2x2 Triangular{Float64,Array{Float64,2},:U,false}:
 1.0  0.0
 0.0  1.0)
)


julia> rand(td)
ERROR: GenericMvNormal not defined
 in _rand! at /home/niedermayer/.julia/v0.3/Distributions/src/multivariate/mvtdist.jl:142
 in rand at /home/niedermayer/.julia/v0.3/Distributions/src/multivariates.jl:18
```

This is fixed by pull request #328

**Comment by afniedermayer** [comment]
It seems that pull request #328 only fixes the typo introduced by a refactoring from GenericMvNormal to MvNormal. But it seems that the original algorithm already contained a bug. Here's an example in which I generate 1 million random draws and do not get back the original covariance matrix:

```julia
using Distributions, PDMats
td=MvTDist(3.,[0.,0.],PDMat([1. 0.;0. 1.]))
x=rand(td,1_000_000);
println(cov(td))
println(cov(x'))
```
which returns
```
[3.0 0.0
 0.0 3.0]
[0.4397361101526044 -0.0013065885945805072
 -0.0013065885945805072 1.6215411206684638]
```

For the univariate t-distribution, random number generation seems to be OK:
```julia
td1=TDist(3.)
y=[rand(td1,1_000_000) rand(td1,1_000_000)]';
println(var(td1))
println(cov(y'))
```
outputs
```
3.0
[2.9357832915809703 -0.00216158423511952
 -0.00216158423511952 2.9300057612313912]
```

**Comment by kmsquire** [comment]
Closed by #334.
